src/metabase/automagic_dashboards/core.clj

```diff
@@ -779,14 +779,14 @@
           [(collect-metrics root question)
            (collect-breakout-fields root question)])))
 
-(defn- preserve-joins
-  "Hack to shove back in joins when they get automagically stripped out by the question decomposition into metrics"
-  [entity dashboard]
-  (if-let [join-statement (get-in entity [:dataset_query :query :joins])]
+(defn- preserve-entity-element
+  "Ensure that elements of an original dataset query are preserved in dashcard queries."
+  [dashboard entity entity-element]
+  (if-let [element-value (get-in entity [:dataset_query :query entity-element])]
     (letfn [(splice-joins [dashcard]
```

Comment:
you should rename `splice-joins` now that it handles joins and expressions. One thing is that it does an `assoc-in`. Are we worried about clobbering things here? Should it be aware that it needs to combine in some manner rather than replace? (its ok if the answer is no right now)

Reply:
+1 on `splice-joins` rename. The old code did an `assoc-in`. I had considered `conj` but I think this is a built-up query so for now it's ok to just replace.

Reply:
Renamed to `splice-elements`.

Reply:
Modified `preserve-entity-element` to preserve any existing element data.